swift/Sources/SignalClient/ContiguousMutableBytes.swift

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright 2020 Signal Messenger, LLC
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+import Foundation
+
+public protocol ContiguousMutableBytes {
+    @inlinable mutating func withUnsafeMutableBytes<R>(_ body: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R
+}
+
+extension Data: ContiguousMutableBytes {}
+extension Array: ContiguousMutableBytes {}
```

Comment:
Honestly this isn't really what we want; like ContiguousBytes we should only conform when the element is UInt8. (Unfortunately Data only has `withUnsafeMutableBytes` and `withContiguousMutableStorageIfAvailable` but not `withContiguousMutableStorageEvenIfNotCurrentlyContiguous`.)

Foundation's MutableDataProtocol is also very close to what we want, but while it gives us a mutable collection and contiguous regions we could feed piecemeal into AES-GCM, it doesn't give us both at once (mutable contiguous regions). I suspect this is an oversight that now can't be corrected for compatibility reasons. :-(

Reply:
Given your API change suggestions, I think we can perhaps do away with this anyhow